_posts/2018-06-18-configuration.md

```diff
@@ -0,0 +1,160 @@
+---
+author: Sean Callan
+author_link: https://github.com/doomspork
+categories: general
+date: 2018-06-18
+layout: post
+title:  Confusing Configuration
+excerpt: >
+  We attempt to clear up some confusion around configuration by looking at the different types, the roles they play, and a different approach we could take.
+---
+
+There's been a lot of discussion about configuration in the community lately.
+We thought this would be an opportune time to discuss configuration and how best to handle it within an Elixir application.
+It is surprising to see how a small change to our applications configuration can eliminate much of the headaches others are experiencing._
+
+### Configuration types
+
+Before we go any further, let's look at the two configuration types and the roles they play.
+
+__Runtime Configuration__
+
+If you've ever used a system environment variable to configure some part of your application, then you're familiar with runtime configuration.
+As the name suggests, this is the configuration for an application at the time it is run.
+These are the values we can expect to change as we deploy our build artifacts to different systems.
+
+__Application Configuration__
+
+Application configuration is something different, and the difference, while subtle, can be a pitfall in certain situations.
+The difference shines when we consider that our code, and it's configuration, is compiled into a build artifact we can distribute.
+
+We can say with certainty that no matter where our application is run there are certain things we want to remain constant.
```

Comment:
> where our application is run

You mean a particular, compiled instance of an application, right? I don't think you're making a statement about how the application will run, no matter what the build-time configuration is. "Application" in this article seems to refer to the uncompiled code sometimes, and to a particular instance of compiled code at other times. Since you're assigning a technical meaning to "application configuration," it may be worth assigning a technical meaning to "application," defining it, and using it in only one sense of the word.